linera-views-derive/src/lib.rs

```diff
@@ -35,16 +38,45 @@ fn get_type_field(field: syn::Field) -> Option<syn::Ident> {
     }
 }
 
-fn context_and_constraints(template_vect: &[syn::Ident]) -> (TokenStream2, TokenStream2) {
-    let context = template_vect
-        .get(0)
-        .expect("failed to find the first generic parameter")
-        .into_token_stream();
-    let constraints = quote! {
-        where
-            #context: linera_views::common::Context + Send + Sync + Clone + 'static,
-            linera_views::views::ViewError: From<#context::Error>,
-    };
+fn context_and_constraints(
+    attributes: &[Attribute],
+    template_vect: &[syn::Ident],
+) -> (Type, Option<TokenStream2>) {
+    let maybe_specific_context = attributes
+        .iter()
+        .find(|attribute| attribute.path.is_ident("specific_context"));
+    let context;
+    let constraints;
+
+    if let Some(specific_context) = maybe_specific_context {
+        match specific_context.parse_meta() {
+            Ok(Meta::NameValue(MetaNameValue {
+                lit: Lit::Str(value),
+                ..
+            })) => {
+                context = value.parse().expect("Invalid specific context");
+                constraints = None;
+            }
+            _ => panic!(
+                r#"Invalid `specific_context` attribute syntax. \
```

Comment:
I'm tempted to drop `specific_`.
By analogy with `#[serde(rename = "Bla")]`, shouldn't it be `#[view(context = "Foo")]` ?

Reply:
Done.